django/db/migrations/loader.py

```diff
@@ -316,6 +319,33 @@ def _reraise_missing_dependency(migration, missing, exc):
                         # "child" is not in there.
                         _reraise_missing_dependency(migration, child, e)
 
+    def check_consistent_history(self, connection):
+        """
+        Checks that all migrations are applied after their dependencies. Raises
+        InconsistentMigrationHistory exception when migration in applied before
+        it's dependency
+        """
+        recorder = MigrationRecorder(connection)
+        applied = recorder.applied_migrations()
+        for migration in applied:
+            # If we don't know anything about this migration we skip it
+            if migration not in self.graph.nodes:
+                continue
+            unapplied_parents = [
+                x.key for x in self.graph.node_map[migration].parents
+                if x not in applied and settings.MIGRATION_MODULES.get(x[0])
```

Comment:
I think what you want is this:

```python
unapplied_parents = [
    x.key for x in self.graph.node_map[migraiton].parents
    if x not in applied and (
        x[0] in settings.MIGRATION_MODULES and settings.MIGRATION_MODULES[x[0]] is not None
    )
]
```

where the condition could be changed to

```python
if x not applied and settings.MIGRATION_MODULES.get(x[0], object) is None
```

You probably want to add a comment for that, though, especially for the second code snippet ;)